Lib/test/test_dataclasses.py

```diff
@@ -3672,6 +3681,36 @@ def test_no_types(self):
                                              'y': int,
                                              'z': 'typing.Any'})
 
+    def test_module_attr(self):
+        self.assertEqual(ByMakeDataClass.__module__, __name__)
+        self.assertEqual(ByMakeDataClass(1).__module__, __name__)
```

Comment:
Is there particular logic in dataclasses related to accessing `__module__` on an instance? If not, it seems like this line is just checking Python's general behavior around accessing class attributes via instance, and probably isn't needed?

Reply:
Dataclasses have this class/instance attributes mix up by their nature. So, I wanted to be double sure here.